crates/common/tedge_utils/src/inotify.rs

```diff
@@ -0,0 +1,17 @@
+pub use inotify_crate::EventMask;
+pub use inotify_crate::WatchMask;
+use inotify_crate::{EventStream, Inotify};
+use std::path::Path;
+
+pub fn inofity_file_watch_stream(
+    config_file: &Path,
+    watch_flags: WatchMask,
+) -> Result<EventStream<[u8; 1024]>, anyhow::Error> {
+    let buffer = [0; 1024];
+    let mut inotify = Inotify::init().expect("Error while initializing inotify instance");
```

Comment:
What can cause this error? Don't forget that `expect` panics !

Reply:
It can be caused by: 
1. ~invalid value specified in flags~ (not relevant as the crate abstracts this away from the user)
2. maximum number of inotify instances has been reached
3. maximum system limit of file descriptors 
4. insufficient kernel memory available

See [this](https://linux.die.net/man/2/inotify_init1).